centos.sh

```diff
@@ -1,83 +1,85 @@
-      echo -e "# Gateway is set in separate file" >> $CONFIGFILE 2>>$DEBUGFILE
-      echo -e "# Do not forget to change interface in file route-$1 and rename this file" >> $CONFIGFILE 2>>$DEBUGFILE
+      echo -e "# Note for customers who want to create bridged networking for virtualisation:" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
+      echo -e "# Gateway is set in separate file" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
+      echo -e "# Do not forget to change interface in file route-$1 and rename this file" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
     fi
-    echo -e "#" >> $CONFIGFILE 2>>$DEBUGFILE
-    echo -e "# device: $1" >> $CONFIGFILE 2>>$DEBUGFILE
-    echo -e "DEVICE=$1" >> $CONFIGFILE 2>>$DEBUGFILE
-    echo -e "BOOTPROTO=none" >> $CONFIGFILE 2>>$DEBUGFILE
-    echo -e "ONBOOT=yes" >> $CONFIGFILE 2>>$DEBUGFILE
-
-    if [ "$3" -a "$4" -a "$5" -a "$6" -a "$7" ]; then
-      echo -e "HWADDR=$upper_mac" >> $CONFIGFILE 2>>$DEBUGFILE
-      echo -e "IPADDR=$3" >> $CONFIGFILE 2>>$DEBUGFILE
+    echo -e "#" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
+    echo -e "# device: $1" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
+    echo -e "DEVICE=$1" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
+    echo -e "BOOTPROTO=none" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
+    echo -e "ONBOOT=yes" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
+
+    if [ -n "$3" ] && [ -n "$4" ] && [ -n "$5" ] && [ -n "$6" ] && [ -n "$7" ]; then
+      echo -e "HWADDR=$upper_mac" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
+      echo -e "IPADDR=$3" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
       if is_private_ip "$3"; then
-        echo -e "NETMASK=$5" >> $CONFIGFILE 2>>$DEBUGFILE
-        echo -e "GATEWAY=$6" >> $CONFIGFILE 2>>$DEBUGFILE
+        echo -e "NETMASK=$5" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
+        echo -e "GATEWAY=$6" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
       else
-        echo -e "NETMASK=255.255.255.255" >> $CONFIGFILE 2>>$DEBUGFILE
-        echo -e "SCOPE=\"peer $6\"" >> $CONFIGFILE 2>>$DEBUGFILE
-
-        echo -e "### Hetzner Online GmbH - installimage" > $ROUTEFILE 2>>$DEBUGFILE
-        echo -e "# routing for eth0" >> $ROUTEFILE 2>>$DEBUGFILE
-        echo -e "ADDRESS0=0.0.0.0" >> $ROUTEFILE 2>>$DEBUGFILE
-        echo -e "NETMASK0=0.0.0.0" >> $ROUTEFILE 2>>$DEBUGFILE
-        echo -e "GATEWAY0=$6" >> $ROUTEFILE 2>>$DEBUGFILE
+        echo -e "NETMASK=255.255.255.255" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
+        echo -e "SCOPE=\"peer $6\"" >> "$CONFIGFILE" 2>> "$DEBUGFILE"
+
+        echo -e "### $COMPANY - installimage" > "$ROUTEFILE" 2>> "$DEBUGFILE"
+        echo -e "# routing for eth0" >> "$ROUTEFILE" 2>> "$DEBUGFILE"
+        echo -e "ADDRESS0=0.0.0.0" >> "$ROUTEFILE" 2>> "$DEBUGFILE"
+        echo -e "NETMASK0=0.0.0.0" >> "$ROUTEFILE" 2>> "$DEBUGFILE"
+        echo -e "GATEWAY0=$6" >> "$ROUTEFILE" 2>> "$DEBUGFILE"
       fi
     fi
 
-    if [ "$8" -a "$9" -a "${10}" ]; then
+    if [ -n "$8" ] && [ -n "$9" ] && [ -n "${10}" ]; then
```

Comment:
after removing:
centos.sh|66 col 44 error| Braces are required for positionals over 9, e.g. ${10}. [SC1037]